diff --git a/include/flatbuffers/base.h b/include/flatbuffers/base.h
index 458ac3f7..b13dcbdb 100644
--- a/include/flatbuffers/base.h
+++ b/include/flatbuffers/base.h
@@ -32,7 +32,7 @@
 #include <cstdlib>
 #include <cstring>
 
-#if defined(ARDUINO) && !defined(ARDUINOSTL_M_H)
+#if defined(ARDUINO) && !defined(ARDUINOSTL_M_H) && defined(__AVR__)
   #include <utility.h>
 #else
   #include <utility>
